framework/src/actions/AddVariableAction.C

```diff
@@ -168,15 +168,19 @@ AddVariableAction::act()
   addVariable(var_name);
 
   // Set the initial condition
-  auto value = _pars.get<std::vector<Real>>("initial_condition");
   if (_pars.isParamValid("initial_condition"))
-    createInitialConditionAction(var_name, value);
+  {
+    auto value = _pars.get<std::vector<Real>>("initial_condition");
+    createInitialConditionAction(value);
+  }
 }
 
 void
-AddVariableAction::createInitialConditionAction(const std::string var_name,
-                                                const std::vector<Real> & value)
+AddVariableAction::createInitialConditionAction(const std::vector<Real> & value)
 {
+  // Variable name
+  std::string var_name = name();
```

Comment:
You don't want the name of the action though right? I was expecting a new `virtual std::string varName() const`